external-crates/move/crates/move-compiler/src/hlir/translate.rs

```diff
@@ -533,7 +527,9 @@ fn constant(context: &mut Context, _name: ConstantName, cdef: T::Constant) -> H:
         value: tvalue,
     } = cdef;
     context.push_warning_filter_scope(warning_filter);
-    let signature = base_type(context, tsignature);
+    let mut diags = Diagnostics::new();
+    let signature = base_type(&mut diags, tsignature);
+    context.add_diags(diags);
```

Comment:
It seems to me that simply taking the `DiagnosticsReporter` would be much nicer here, as we can just grab it off the context instead of making a dozen of these and ~~hoping~~ ensuring we call `add_diagnostics` every time.